### DISCUSSION TITLE
Template info for merkury MI-WW119-199W

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16682] [General]
https://templates.blakadder.com/merkury_MI-WW119-199W.html
```
VCC and ground available on opposite sides of shorter segment of module sticking through main board when attached. VCC is the wider solder joint and ground is the 1st pad on opposite side. TX and RX available on top edge of module closest pins to the antenna. GND and D0 are available opposite end of top edge, last pad and 4th pad.
```

Has anyone flashed this using serial. Can you share a bit of info on what some of this means.
VCC, is it needed or is ground and TX and RX only needed?

**Comment by rebel7580** [comment]
You need VCC to power up the device with an external DC power source. DO NOT plug this into the mains AC while your flashing it!!

**Comment by dac2020** [comment]
Hi,

Yes, that makes sense. It's what I thought but wasn't sure. 

I guess VCC=5vdc+ and GND=5vdc- and TX to TX and RX to RX as usual. 
Sometimes GND is also needed for serial but most times not from the few things I've done so far at least.

Am I correct to assume this or there are variations?

**Reply by rebel7580** [reply]
You need to connect GND on your device to the ground of your serial flasher too, so they both have a common ground reference.

**Reply by dac2020** [reply]
When you say serial flasher, I assume you mean my USB/TTL device connected using serial.

This is all new but I'm picking it up fairly quickly with the help I'm getting here :).

**Reply by rebel7580** [reply]
> When you say serial flasher, I assume you mean my USB/TTL device connected using serial.
>
Yes.

**Reply by sfromis** [reply]
GND is absolutely always needed.

**Reply by dac2020** [reply]
Got it. Thank you.

**Reply by dac2020** [reply]
Oh, one more thing. Is there a standard baud and flow control on all these kinds of chips?

**Reply by sfromis** [reply]
In a broad perspective, no. But 115200 is very commonly working (including with ESP8266 and ESP32), with 9600 not too rare. Flow control is "usually"  not used, just simple UART serial.

(Some terminal programs can autodetect bit rate; while this sounds nice, it does carry the risk of confusion when other things does not work).

**Reply by dac2020** [reply]
Thanks. I'll note this too.

**Comment by dac2020** [comment]
Ok, first time ever doing this.
I soldered pins to the tiny SOC. I have power and RX/TX/GND connected and the device has green LED.
At either 9600 or 115200, there is no output what so ever. 
I assume I should be seeing something.

**Reply by sfromis** [reply]
You can't bank on that, unless you know the device is in a situation where you should expect it to send something. Note that you connect tx/transmit on one end to rx/receive on the other end, not straight.

**Reply by dac2020** [reply]
Terminal TX is connected to RX and RX to TX, GND to GND.

**Reply by dac2020** [reply]
The device is fully pulled out of the plug.
![photo_2022-09-29_14-50-41](https://user-images.githubusercontent.com/9727841/193149153-21e6ac78-ce62-4239-a554-6dbffa9400e5.jpg)

**Comment by dac2020** [comment]
What does it mean I can't bank on that? How do I flash the device if I don't get any output from it?

**Reply by sfromis** [reply]
I already explained "can't bank on that":
> unless you know the device is in a situation where you should expect it to send something

Point being that devices cannot be expected to send continuously, but only in certain situations. ESP8266 will send briefly after boot, after that everything is up to the firmware in use. I've seen some firmwares being chatty on serial during operation, but not as a general expectation.

What I usually do is to hookup my wiring as I expect to to be correct, and verify by running esptool.py with flash_id after having booted with gpio0 low during power up (or run whatever other flashing tool). If that works, success, if not, start looking.

**Comment by dac2020** [comment]
I read what you said and understood. 
The laptop Im using to test is Windows so limited in tools. I'll have to pull out the Linux one next so I have more tools. 
I'm not getting a single character out of this board. It does have a small switch mounted directly onto it, presumably to flash or reset.

```
What I usually do is to hookup my wiring as I expect to to be correct, and verify by running esptool.py with flash_id after having booted with gpio0 low during power up (or run whatever other flashing tool). If that works, success, if not, start looking.
```
Over the head of someone totally new to this :).
The esptool.py is something I don't know anything about so would have to look into this now. 
Maybe this is more than I can handle then.

**Reply by sfromis** [reply]
Since I **know** that you've seen the getting-started page, "esptool" should not be an unknown concept to you. It is a command-line tool for flashing.

Anyway, the browser-based Tasmota Web Installer is recommended (on the getting-started page) for easy flashing, without even needing to install anything on your OS.

**Reply by barbudor** [reply]
and available for Windows too
I'm using Windows and never felt in lack of tools when developping for home automation
More the opposite I would say

**Reply by dac2020** [reply]
LOL sure, I've seen a lot of pages since starting with this but I sure don't remember it all. Just learning remember.
Tasmota web installer? Nope, don't even remember that so off to look at those pages again :).

**Reply by dac2020** [reply]
Right then, that's one cool tool!

Anyone know how I get this TYWE2S into programming mode? There's a button on it but I can't seem to get the mode to change.

**Reply by sfromis** [reply]
Depends on the button being connected to gpio0. Having a button is useless, unless it works to ground gpio0.

**Reply by dac2020** [reply]
I have no idea since it looks to be a layered board.
Hate to give up but not sure what I can do next.

**Reply by barbudor** [reply]
a multimeter doesn't care if the track is visible or not as far as you can check continuity between GPIO0 on the TYWE2S  and a pin of the button

**Reply by sfromis** [reply]
You can just check if gpio0 connects to that button. Use a multimeter in continuity checking mode. Or just forget about the button and manually connect gpio0 to ground while powering up, you need not solder it for that.

**Reply by dac2020** [reply]
Is it just momentary or during the whole process? Maybe I can hold something down. I think I see the gpio0 pad based on the images.

**Reply by dac2020** [reply]
I would have to order a vise next as I cannot hold this tiny board while trying to use a multi-meter. That was the obvious next conclusion to check but also need to know if I just need to short those for a moment on power up or during the whole process.

**Reply by dac2020** [reply]
Not connected to switch.

**Reply by sfromis** [reply]
Gpio0 is needed momentarily while powering up, thus my addition of no soldering needed for that part. While a vise or extra hands can be handy to have at hand, in a pinch you can do much holding of stuff between your knees.

**Reply by dac2020** [reply]
Haha, yes, I've tried all kinds of ways so far and none work, it's just too small a board. 
I found a vise at least but shorting gnd and gpio0 on boot isn't doing anything different.

**Reply by dac2020** [reply]
Shorting while flashing doesn't work.
Shorting for a moment after powering doesn't work.
Shorting as I power up doesn't work.

**Reply by sfromis** [reply]
Not just boot, but power-up. Connect before applying power.

**Reply by sfromis** [reply]
When things does not work, triple-check your wiring **and** power supply situation. Make no assumptions about anything being right already. Do not ignore the part of power supply status, many USB/serial adapters have unsuitably low ability to deliver 3.3V power.

**Reply by dac2020** [reply]
I've checked and checked and solder, etc all are fine. How long does it take to boot? Maybe I'm not holding it long enough.

**Reply by dac2020** [reply]
This is what I'm using which provides 3.3v.

https://www.tradeling.com/ae-en/product-details/dsd-tech-usb-to-ttl-serial-adapter-clear-22-x-08-x-01inch-sh-u09c-624fde9b308af9001bbf98cf-5e495521f60e52001b9c7631

**Reply by rebel7580** [reply]
I'm not 100% familiar with that flasher, but I dont see a discrete 5->3.3v regulator, so I'm assuming that 3.3v output is coming from the built-in 3.3v reg in the FT232 chip, which is only rated for 50ma MAX. Not enough to power the device IMHO. Better to have an external 3.3 supply.

**Reply by dac2020** [reply]
I don't have one and I've tried two devices. 
It's time to give up. I think I'm done with the hardware hacking at this level as it's too time consuming.
Much of my time has been spent trying to find clips and ways to hold these tiny things and I would have to keep ordering stuff to do this work regularly.

I'll try again when I have some time but for now, I'll start looking for hardware that's ready to go or easier to convert.

Thanks for all the help.
